### DISCUSSION TITLE
200 Response without body and content-type

### DISCUSSION THREAD

**Discussion by bunny-therapist** [litestar-org/litestar#3097] [Q&A] [answered]
How can I make an endpoint return a Response without a body and without a content-type? Returning `None` from the route handler just gets me "application/json" media type with `null` as the body. I have looked in the documentation but I can't find anything.

**Comment by vkcku** [comment]
@bunny-therapist setting the status code to 204 should work.

```python
from litestar import get
from litestar.app import Litestar
from litestar.enums import MediaType
from litestar.testing import create_test_client
from http import HTTPStatus


@get("/", status_code=HTTPStatus.NO_CONTENT)
async def handler() -> None:
    return None


with create_test_client([handler]) as client:
    resp = client.get("/")
    header = resp.headers

    assert "content-type" not in resp.headers
    assert resp.num_bytes_downloaded == 0
```

**Comment by provinzkraut** [comment, marked as answer]
"200 without a body" isn't really a thing. As @guacs said, if you truly want to omit the body, you should use a `204` code. However, if you're fine with sending an empty body, you could just do something like

```python
@get("/")
async def handler() -> str:
    return ""
```

which sends an empty plaintext response.

**Reply by vkcku** [reply]
This will set the content type as text though.

**Reply by bunny-therapist** [reply]
I understand. The only reason I am trying to return 200 without a body is because I am trying to replace existing code. I am currently investigating if 204 would work as well (it should, but you never know).

**Reply by provinzkraut** [reply]
@bunny-therapist What I'm trying to say is: The example I posted should result in the same bytes being sent over the wire as omitting a body. When sending an ASGI response to the server, you have to send at least one "body" event, even if it's an empty string (`b""`), which sending that empty string directly is effectively the same as returning `None`, so your client would see the same body in the end. The only difference is that you'll have to set a `Content-Type`. However, while technically valid, omitting this will result in most clients simply interpreting the result as `text/plain` anyway.

<hr>

*Returning empty string*

```
< HTTP/1.1 200 OK
< date: Mon, 12 Feb 2024 15:09:53 GMT
< server: uvicorn
< content-type: text/plain; charset=utf-8
< content-length: 0
< 
```

*Returning `None`*
```
< HTTP/1.1 204 No Content
< date: Mon, 12 Feb 2024 15:10:01 GMT
< server: uvicorn
< 
```

As you can see, the received response body is the same. The only difference is in the headers.
